concept/answer/ConceptMap.java

```diff
@@ -219,7 +228,25 @@ public static Comparator create(Modifiers.Sorting sorting) {
         private static java.util.Comparator<ConceptMap> variableComparator(Retrievable var) {
             return java.util.Comparator.comparing(
                     (ConceptMap conceptMap) -> conceptMap.get(var), (concept1, concept2) -> {
-                        if (concept1.isAttribute() && concept2.isAttribute()) {
+                        if (concept1.isValue() && concept2.isValue()) {
+                            Value<?> val1 = concept1.asValue();
+                            Value<?> val2 = concept2.asValue();
+                            if (val1.isString()) {
+                                return Encoding.ValueType.STRING.comparator().compare(val1.asString().value(), val2.asString().value());
+                            } else if (val1.isBoolean()) {
+                                return Encoding.ValueType.BOOLEAN.comparator().compare(val1.asBoolean().value(), val2.asBoolean().value());
+                            } else if (val1.isLong() && val2.isLong()) {
+                                return Encoding.ValueType.LONG.comparator().compare(val1.asLong().value(), val2.asLong().value());
+                            } else if (val1.isDouble() || val2.isDouble()) {
+                                Double double1 = val1.isLong() ? val1.asLong().value() : val1.asDouble().value();
+                                Double double2 = val2.isLong() ? val2.asLong().value() : val2.asDouble().value();
+                                return Encoding.ValueType.DOUBLE.comparator().compare(double1, double2);
+                            } else if (val1.isDateTime()) {
+                                return Encoding.ValueType.DATETIME.comparator().compare(val1.asDateTime().value(), val2.asDateTime().value());
+                            } else {
+                                throw TypeDBException.of(ILLEGAL_STATE);
+                            }
+                        } else if (concept1.isAttribute() && concept2.isAttribute()) {
```

Comment:
Adds the case of value variables. 
(The diff is a bit unintuitive because I added Values as the first case)

Reply:
We could see if we can go even further and make an interface for values themselves.